Assets/Scripts/Framework/Train/TrainDoorsAnimator.cs

```diff
@@ -0,0 +1,35 @@
+using UnityEngine;
+
+namespace DerailedDeliveries.Framework.Train
+{
+    /// <summary>
+    /// Class responsible for opening and closing train doors.
+    /// </summary>
+    public class TrainDoorsAnimator : MonoBehaviour
+    {
+        private int _doorsOpenAnimationHash;
+        private int _doorsCloseAnimationHash;
+
+        private Animator[] _doorAnimators;
+
+        private void Awake() => _doorAnimators = GetComponentsInChildren<Animator>();
```

Comment:
I think it is better to keep this script on the same object as the animator. If we use this approach with `GetComponentsInChildren` we will get errors when adding other animators on the train.